src/main/java/org/apache/sling/feature/maven/mojos/IncludeArtifactMojo.java

```diff
@@ -90,12 +138,29 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             ProjectHelper.getFeatures(this.project).put(key, found);
             ProjectHelper.getAssembledFeatures(this.project).put(key, found);
         }
+        getLog().debug("Found Feature: " + found + ", artifact: " + art);
         includeArtifact(found, includeArtifactExtension, art);
+        getLog().debug("Feature Key: " + key + ", feature from key: " + ProjectHelper.getAssembledFeatures(this.project).get(key));
         includeArtifact(ProjectHelper.getAssembledFeatures(this.project).get(key), includeArtifactExtension,
                 art.copy(art.getId()));
+
+        addDependencies(found);
+
+        // Obtain any features from Source folder and add any Extensions to the target feature
+        final FeatureSelectionConfig featureSelectionConfig = new FeatureSelectionConfig();
+        featureSelectionConfig.setFilesInclude("**/*.json" );
+        if(file != null) {
+            featureSelectionConfig.setFilesExclude("**/" + file.getName());
+        }
+        final Map<String, Feature> selection = this.getSelectedFeatures(featureSelectionConfig);
+
+        includeFeatures(selection, found);
+
+        // Write the Feature into its file and install it
         if (file != null) {
             try ( final Writer writer = new FileWriter(file)) {
                 FeatureJSONWriter.write(writer, found);
+                installFMDescriptor(file, found);
```

Comment:
Attaching of the artifact should not be part of this mojo - the feature is added to the project artifacts and therefore the AttachFeaturesMojo will attach it

Reply:
After some back and forth I was able to install the FM using the Attach Features Mojo and so I am going to remove this.